modules/apps/object/object-service/src/main/java/com/liferay/object/internal/upgrade/v8_8_2/ObjectFolderGuestPermissionUpgradeProcess.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.object.internal.upgrade.v8_8_2;
+
+import com.liferay.object.model.ObjectFolder;
+import com.liferay.object.service.ObjectFolderLocalService;
+import com.liferay.portal.kernel.model.ResourceConstants;
+import com.liferay.portal.kernel.model.Role;
+import com.liferay.portal.kernel.model.role.RoleConstants;
+import com.liferay.portal.kernel.security.permission.ActionKeys;
+import com.liferay.portal.kernel.service.CompanyLocalService;
+import com.liferay.portal.kernel.service.ResourcePermissionLocalService;
+import com.liferay.portal.kernel.service.RoleLocalService;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+
+/**
+ * @author Guilherme Sa
+ */
+public class ObjectFolderGuestPermissionUpgradeProcess extends UpgradeProcess {
+
+	public ObjectFolderGuestPermissionUpgradeProcess(
+		CompanyLocalService companyLocalService,
+		ObjectFolderLocalService objectFolderLocalService,
+		ResourcePermissionLocalService resourcePermissionLocalService,
+		RoleLocalService roleLocalService) {
+
+		_companyLocalService = companyLocalService;
+		_objectFolderLocalService = objectFolderLocalService;
+		_resourcePermissionLocalService = resourcePermissionLocalService;
+		_roleLocalService = roleLocalService;
+	}
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		_companyLocalService.forEachCompany(
+			company -> {
+				Role guestRole = _roleLocalService.getRole(
+					company.getCompanyId(), RoleConstants.GUEST);
+
+				ObjectFolder objectFolder =
+					_objectFolderLocalService.getUncategorizedObjectFolder(
+						company.getCompanyId());
+
+				_resourcePermissionLocalService.setResourcePermissions(
+					company.getCompanyId(), ObjectFolder.class.getName(),
+					ResourceConstants.SCOPE_INDIVIDUAL,
+					String.valueOf(objectFolder.getObjectFolderId()),
+					guestRole.getRoleId(), new String[] {ActionKeys.VIEW});
```

Comment:
Could you put [this code ](https://github.com/liferay-objects/liferay-portal/pull/2999/commits/faf335f3ce30e91526c3eb8e4beb47f693148b79#diff-97cf5678725d873430ba3e1fa4b340dfe971f973360c32c31be78a43274896bdR40-R51) into a try catch?